src/pudl_archiver/archivers/validate.py

```diff
@@ -0,0 +1,35 @@
+"""Defines models used for validating/summarizing an archiver run."""
+from typing import Literal
+
+from pydantic import BaseModel
+
+
+class ValidationTestResult(BaseModel):
+    """Class containing results of a validation test, and metdata about the test."""
+
+    name: str
+    description: str
+    ignore_failure: bool = False
+    resource_name: str | None = None
+    result: bool
+    note: str | None = None
+
+
+class FileDiff(BaseModel):
+    """Model summarizing changes to a single file in a deposition."""
+
+    name: str
+    diff_type: Literal["CREATE", "UPDATE", "DELETE"]
+    size_diff: int
+
+
+class RunSummary(BaseModel):
```

Comment:
I started outlining a report that we could generate for every run that would include a summary of the changes, and results of any tests.